src/nspawn/nspawn-mount.c

```diff
@@ -916,11 +916,12 @@ int mount_cgroups(
                 const char *dest,
                 bool unified_requested,
                 bool userns, uid_t uid_shift, uid_t uid_range,
-                const char *selinux_apifs_context) {
+                const char *selinux_apifs_context,
+                bool use_cgns) {
 
         if (unified_requested)
                 return mount_unified_cgroups(dest);
-        else if (cg_ns_supported())
+        else if (cg_ns_supported() && use_cgns)
```

Comment:
Nitpick: invert `use_cgns` with `cg_ns_supported()`.Due to "if lazy evaluation", if `use_cgns` is 0 `cg_ns_supported()` it is not even called.